### PR TITLE
[test][devx] modify base runner to work with rde and retool specific concepts

### PR DESCRIPTION
**Pull request by TumuGuskun**
this modifies an existing extension to run the correct `rde` commands with the correct flags to test describe and it blocks. It only shows up in the backend and packages folders. There's a bunch of stuff we don't need because rde takes care of it

see it in action here:
https://www.loom.com/share/4c725802eac04f119dc4f4cdc0ee71c0?sid=28c25f1a-1c96-46d9-84ff-acf0d97b7fb2

### CODE REVIEW

File: images/icon.jpeg
Comment (rkwon-retool) [inline images/icon.jpeg:1]: I appreciate this change. +1

File: package.json [hunk 1]
```diff
@@ -3,18 +3,14 @@
   "displayName": "Bazel Jest Runner",
   "description": "Run Jest test with Bazel in VSCode",
   "version": "0.1.0",
-  "files": [
-    "out",
-    "images"
-  ],
-  "publisher": "mscolnick",
+  "publisher": "TumuGuskun",
```
Comment (rkwon-retool) [inline package.json:6]: Got it. So `mscolnick`'s package is the base from which we're building our vscode extension.

Have we double checked the licensing here? I'd feel a bit icky if we were stepping on this person's intellectual property.

**Review by mertbozfakioglu — APPROVED** [review]
Nice!

**Review by rkwon-retool — APPROVED** [review]
1 major concern. Not related to functionality.

I would also ask how we expose or install this package extension locally! This should also be documented somewhere for other people to get if we're comfortable with it :)

File: package.json [hunk 2]
```diff
@@ -3,18 +3,14 @@
...
   "engines": {
     "vscode": "^1.59.0"
   },
-  "icon": "images/icon.png",
+  "icon": "images/icon.jpeg",
```
Comment (samsternatretool) [inline package.json:10]: Ah a man of culture! Using the `e` in `jpeg`

File: package.json [hunk 3]
```diff
@@ -3,18 +3,14 @@
...
   "repository": {
     "type": "git",
-    "url": "https://github.com/mscolnick/vscode-bazel-jest.git"
+    "url": "https://github.com:tryretool/vscode-bazel-jest.git"
```
Comment (samsternatretool) [inline package.json]: Should `:` be `/`?

File: src/codelensprovider.ts [hunk 4]
```diff
@@ -51,7 +51,9 @@ export class CodeLensProvider implements vscode.CodeLensProvider {
       const text = document.getText();
       const parseResults = parse(document.fileName, text).root.children ?? [];
       const codeLens: CodeLens[] = [];
-      parseResults.forEach((parseResult) => codeLens.push(...getTestsBlocks(parseResult, parseResults)));
+      if (document.uri.fsPath.includes("backend/") || document.uri.fsPath.includes("packages/")) {
```
Comment (samsternatretool) [inline src/codelensprovider.ts:54]: I guess we should warn people that in most `packages` this won't work, but I understand why you're not hardcoding that in the extension.

**Review by samsternatretool — APPROVED** [review]
This is SLICK

File: package.json (hunk 1 above)
Comment (TumuGuskun) [reply package.json:6]: hmm I assumed there was an MIT license but you're right there's no license there. I can reach out to them/put more creds in

File: src/codelensprovider.ts (hunk 4 above)
Comment (TumuGuskun) [reply src/codelensprovider.ts:54]: yeah that's something we can put in rde

**Comment by TumuGuskun** [comment]
> 1 major concern. Not related to functionality.
> 
> I would also ask how we expose or install this package extension locally! This should also be documented somewhere for other people to get if we're comfortable with it :)

@rkwon I'll add stuff to the README for installing locally. I think easiest would be putting it on the marketplace (maybe privately?) and just adding it to the recommended extensions. We have one other extension that does that.

**Pull request merged**